crates/bevy_core_pipeline/src/overlay/overlay_node.rs

```diff
@@ -0,0 +1,147 @@
+use bevy_ecs::prelude::{FromWorld, QueryState, With, World};
+use bevy_math::Vec4;
+use bevy_render::{
+    render_graph::{Node, NodeRunError, RenderGraphContext, SlotInfo, SlotType},
+    render_phase::TrackedRenderPass,
+    render_resource::{
+        encase, BindGroup, BindGroupDescriptor, BindGroupEntry, Buffer, BufferInitDescriptor,
+        BufferUsages, CachedRenderPipelineId, LoadOp, Operations, PipelineCache,
+        RenderPassDescriptor, ShaderSize,
+    },
+    renderer::{RenderContext, RenderDevice, RenderQueue},
+    view::ViewTarget,
+};
+
+use super::{pipeline::OverlayPipeline, CameraOverlay, OverlayDiagnostics};
+
+pub(crate) struct DiagnosticOverlayBuffer {
+    buffer: Buffer,
+    bind_group: BindGroup,
+}
+
+pub(crate) mod graph {
+    pub const NAME: &str = "OVERLAY";
+    pub const NODE: &str = "OVERLAY_PASS";
+    pub const NODE_INPUT: &str = "OVERLAY_PASS_VIEW";
+    pub const IN_VIEW: &str = "OVERLAY_IN_VIEW";
+}
+
+impl FromWorld for DiagnosticOverlayBuffer {
+    fn from_world(world: &mut World) -> Self {
+        let render_device = world.get_resource::<RenderDevice>().unwrap();
+        let overlay_pipeline = world.get_resource::<OverlayPipeline>().unwrap();
+
+        let byte_buffer = [0u8; Vec4::SIZE.get() as usize];
+        let mut buffer = encase::UniformBuffer::new(byte_buffer);
+        buffer.write(&[Vec4::ZERO]).unwrap();
+        let diagnostics_buffer = render_device.create_buffer_with_data(&BufferInitDescriptor {
+            label: Some("diagnostics Buffer"),
+            contents: buffer.as_ref(),
+            usage: BufferUsages::UNIFORM | BufferUsages::COPY_DST,
+        });
+        let diagnostics_bind_group = render_device.create_bind_group(&BindGroupDescriptor {
+            layout: &overlay_pipeline.layout[0],
+            entries: &[BindGroupEntry {
+                binding: 0,
+                resource: diagnostics_buffer.as_entire_binding(),
+            }],
+            label: Some("diagnostics_bind_group"),
+        });
+
+        DiagnosticOverlayBuffer {
+            buffer: diagnostics_buffer,
+            bind_group: diagnostics_bind_group,
+        }
+    }
+}
+
+impl DiagnosticOverlayBuffer {
+    pub(crate) fn write_buffer(
+        &self,
+        diagnostics: &OverlayDiagnostics,
+        render_queue: &RenderQueue,
+    ) {
+        let byte_buffer = [0u8; Vec4::SIZE.get() as usize];
+        let mut buffer = encase::UniformBuffer::new(byte_buffer);
```

Comment:
Don't create a new `UniformBuffer` every frame that updates. Keep a hold of the existing one and update it.